src/Form/RestOaiPmhSettingsForm.php

```diff
@@ -145,34 +145,20 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     $form['mapping'] = [
       '#type' => 'details',
       '#open' => TRUE,
-      '#title' => $this->t('Metadata Mapping'),
+      '#title' => $this->t('Metadata Mappings'),
       '#attributes' => ['style' => 'max-width: 750px'],
-      '#description' => $this->t('<p>Select how the entities that are exposed to OAI-PMH will be mapped to Dublin Core and printed on your OAI-PMH endpoint. There are a few supported options:</p>
-        <p>
-        <ul>
-          <li><strong>Drupal Core RDF Module</strong> if you have Drupal\'s core RDF module enabled, and have an RDF mapping with some Dublin Core elements for your entitities, that field mapping will be used<br><br></li>
-
-          <li><strong>Metatag Dublin Core Module</strong> If you have the <a href="http://drupal.org/project/metatag">Metatag Dublin Core</a> module enabled, and have some Dublin Core properites defined for the entities exposed to OAI-PMH, that mapping will be used.<br><br></li>
-
-          <li><strong>Custom</strong> with a custom module or theme, you can implement<br><code>hook_preprocess_rest_oai_pmh_record</code>,<br>and/or override the <code>rest-oai-pmh-record.html.twig</code> template file</li>
-        </ul></p>'),
+      '#description' => $this->t('<p>Select which metadata plugins should be enabled.</p>'),
     ];
-    $options = [];
-    if ($this->moduleHandler->moduleExists('rdf')) {
-      $options['rdf'] = $this->t('Drupal Core RDF Module');
-    }
-    if ($this->moduleHandler->moduleExists('metatag_dc')) {
-      $options['metatag_dc'] = $this->t('Metatag Dublin Core Module');
+    $mapping_plugin_definitions = \Drupal::service('plugin.manager.oai_metadata_map')->getDefinitions();
+    $mapping_plugin_options = [];
+    foreach ($mapping_plugin_definitions as $plugin_id => $plugin_definition) {
+      $mapping_plugin_options[$plugin_id] = $plugin_definition['label']->render();
     }
-    $options += [
-      'custom' => $this->t('Custom'),
-    ];
-    $form['mapping']['mapping_source'] = [
-      '#type' => 'select',
-      '#title' => $this->t('Metadata Mapping Source'),
-      '#options' => $options,
-      '#required' => TRUE,
-      '#default_value' => $config->get('mapping_source') ? : NULL,
+    $form['mapping']['mapping_plugins'] = [
+      '#type' => 'checkboxes',
```

Comment:
Should this form element be of type `checkboxes`, which allows for multiple metadata mapping formats to be selected? Or of type `select` to only allow for one mapping source. I only envisioned a site admin using one mapping source, but perhaps there's a need for multiple sources?

Also, I think it should be a required form element?